pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -99,22 +99,23 @@ type metadataCerts struct {
 // ClusterBuilder interface provides an abstraction for building Envoy Clusters.
 type ClusterBuilder struct {
 	// Proxy related information used to build clusters.
-	serviceInstances  []*model.ServiceInstance // Service instances of Proxy.
-	metadataCerts     *metadataCerts           // Client certificates specified in metadata.
-	clusterID         string                   // Cluster in which proxy is running.
-	proxyID           string                   // Identifier that uniquely identifies a proxy.
-	proxyVersion      string                   // Version of Proxy.
-	proxyType         model.NodeType           // Indicates whether the proxy is sidecar or gateway.
-	sidecarScope      *model.SidecarScope      // Computed sidecar for the proxy.
-	passThroughBindIP string                   // Passthrough IP to be used while building clusters.
-	supportsIPv4      bool                     // Whether Proxy IPs has IPv4 address.
-	supportsIPv6      bool                     // Whether Proxy IPs has IPv6 address.
-	hbone             bool                     // Does the proxy support HBONE
-	locality          *core.Locality           // Locality information of proxy.
-	proxyLabels       map[string]string        // Proxy labels.
-	proxyView         model.ProxyView          // Proxy view of endpoints.
-	proxyIPAddresses  []string                 // IP addresses on which proxy is listening on.
-	configNamespace   string                   // Proxy config namespace.
+	serviceInstances        []*model.ServiceInstance // Service instances of Proxy.
+	metadataCerts           *metadataCerts           // Client certificates specified in metadata.
+	clusterID               string                   // Cluster in which proxy is running.
+	proxyID                 string                   // Identifier that uniquely identifies a proxy.
+	proxyVersion            string                   // Version of Proxy.
+	proxyType               model.NodeType           // Indicates whether the proxy is sidecar or gateway.
+	sidecarScope            *model.SidecarScope      // Computed sidecar for the proxy.
+	passThroughBindIP       string                   // Passthrough IP to be used while building clusters.
+	extraPassThroughBindIPs []string
```

Comment:
```suggestion
	passThroughBindIP       []string                   // Passthrough IP to be used while building clusters.
```

I am not sure why we need to store them split

Reply:
and can just make getPassthroughBindIps respect EnableDualStack so we don't need to understand the semantics of the passthrough IPs in every use

Reply:
okay, I move the `EnableDualStack` into `getPassthroughBindIps`